code/procs/mobprocs/typing_indicator.dm

```diff
@@ -0,0 +1,78 @@
+#define TYPING_OVERLAY_KEY "speech_bubble"
+
+///Singletons for typing indicators
+var/mutable_appearance/living_speech_bubble = mutable_appearance('icons/mob/mob.dmi', "speech")
+var/mutable_appearance/living_typing_bubble = mutable_appearance('icons/mob/mob.dmi', "typing")
+/mob/proc/create_typing_indicator()
+	return
+
+/mob/proc/remove_typing_indicator()
+	return
+
+/mob/Logout()
+	remove_typing_indicator()
+	. = ..()
+
+////Typing verbs////
+//Those are used to show the typing indicator for the player without waiting on the client.
+
+/*
+Some information on how these work:
+The keybindings for say and me have been modified to call start_typing and immediately open the textbox clientside.
+Because of this, the client doesn't have to wait for a message from the server before opening the textbox, the server
+knows immediately when the user pressed the hotkey, and the clientside textbox can signal success or failure to the server.
+
+When you press the hotkey, the .start_typing verb is called with the source ("say" or "me") to show the typing indicator.
+When you send a message from the custom window, the appropriate verb is called, .say or .me
+If you close the window without actually sending the message, the .cancel_typing verb is called with the source.
+
+Both the say/me wrappers and cancel_typing remove the typing indicator.
+*/
+
+/// Show the typing indicator. The source signifies what action the user is typing for.
+/mob/verb/start_typing(source as text) // The source argument is currently unused
```

Comment:
why do we have params if they're unused for both start and cancel

Reply:
My guess is it's for if anyone wants to use the text arguments for something without needing to go and modify the hellscape that is the winset fuckery.